example_cluster/setup_fake_config.py

```diff
@@ -0,0 +1,57 @@
+import json
+import os
+import shutil
+import socket
+import subprocess
+import sys
+from distutils.dir_util import copy_tree
+
+
+def find_open_ports():
+    for port in range(10000, 1, -1):
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            if sock.connect_ex(('localhost', port)) != 0:
+                return port
+
+
+def main():
+
+    print('-------------------------------------------------------')
+    print('Please run echo $$PAASTA_SYSTEM_CONFIG_DIR to continue')
+    print("Please set environment variable PAASTA_TEST_CLUSTER to the cluster you want to use.")
+    print("This is necessary for tron jobs")
+    print('-------------------------------------------------------')
+    cluster = os.environ.get('PAASTA_TEST_CLUSTER', 'norcal-devc')
+    config_path = 'fake_config'
+
+    copy_tree('/etc/paasta', config_path)
+    # Generate tron.json
+    tron_config = {'tron': {'url': f'http://tron-{cluster}:8089'}}
+    with open(config_path + '/tron.json', 'w') as f:
+        json.dump(tron_config, f)
+    # find unused port
+    port = find_open_ports()
+    # Generate api endpoints
+    api_endpoints = {'api_endpoints': {cluster: f'http://localhost:{port}'}}
+    os.remove(config_path + '/api_endpoints.json')
+    with open(config_path + '/api_endpoints.json', 'w') as f:
+        json.dump(api_endpoints, f)
+
+    try:
+        # export config path
+        os.environ['PAASTA_SYSTEM_CONFIG_DIR'] = config_path
+        # execute paasta api command
+        p = subprocess.Popen(
```

Comment:
How about just `exec`?

Reply:
Isn't exec used to execute python code? I'll need to manually parse arguments then.

Reply:
That is `eval`.  Exec makes it so a new process "takes over"

https://docs.python.org/3/library/os.html#process-management

Reply:
I'm not sure if I can do it. `I tried os.execl('.tox/py36-linux/bin/python', 'python', '-m', 'paasta_tools.api.api', *['-D', '-c', cluster, str(port)])` which again does not have the installed dependency.